### ISSUE TITLE
py_test allows non-hermetic unspecified dependencies

### ISSUE THREAD

**Issue by mdvorsky** [bazelbuild/bazel#8501]
### Description of the problem / feature request:

py_test allows the source file to import an unspecified dependency

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
$ cat WORKSPACE
$ cat BUILD
py_test(
    name = "test",
    srcs = ["test.py"]
)
$ cat test.py
import import_test
print('pass')
$ cat import_test.py
$ bazel test test
...
//:test                            PASSED in 0.1s
```

### What operating system are you running Bazel on?

macOS Mojave 10.14.5

### What's the output of `bazel info release`?

release 0.25.3-homebrew

###  Have you found anything relevant by searching the web?

No.

**Comment by brandjon** [comment]
Thanks for the report. This is #7091. In order to fix it we'd have to inject some `sys.path` manipulation logic into the initialization of the user Python process (not the stub script), or generate a dummy directory holding only a copy of the main file and nothing else, or both.

**Comment by mdvorsky** [comment]
@brandjon the linked issue https://github.com/bazelbuild/bazel/issues/7091 is now reported fixed in Bazel 1.2.1, but the above repro still works even with that version.

```
$ bazel version
Build label: 1.2.1-homebrew
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Wed Nov 27 03:51:56 2019 (1574826716)
Build timestamp: 1574826716
Build timestamp as int: 1574826716

$ bazel test test
...
//:test                                                                  PASSED in 0.3s
```
